packages/xo-web/src/xo-app/proxies/index.js

```diff
@@ -114,21 +109,110 @@ const COLUMNS = [
     itemRenderer: proxy => <Vm id={proxy.vmUuid} link />,
     name: _('vm'),
   },
+  {
+    itemRenderer: (proxy, { updatesByProxy, upgradeAppliance }) => {
+      const globalState = updatesByProxy[proxy.id]
+      if (globalState === undefined) {
+        return
+      }
+
+      const { state } = globalState
+      const upgrade = (
+        <ActionButton
+          btnStyle='success'
+          disabled={proxy.vmUuid === undefined}
+          handler={upgradeAppliance}
+          handlerParam={proxy.id}
+          icon='upgrade'
+        >
+          {_('upgrade')}
+        </ActionButton>
+      )
+
+      if (state.endsWith('-upgrade-needed')) {
+        return (
+          <div>
+            {upgrade}
+            <p className='text-warning'>
+              <Icon icon='alarm' />
+              &nbsp;{_('upgradeAvailable')}
+            </p>
+          </div>
+        )
+      }
+
+      if (
+        state === 'xoa-up-to-date' ||
+        state === 'xoa-upgraded' ||
+        state === 'updater-upgraded' ||
+        state === 'installer-upgraded'
```

Comment:
Where did you find these values?

Reply:
See https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-web/src/common/xoa-updater.js#L129-L146

Reply:
These replies can only occur after upgrading, not when fetching the state.

Reply:
These values cannot happen when fetching state on upgrading a proxy?

Reply:
Indeed, they may happen when fetching state while an upgrade is ongoing.

But in that case it's not good enough to say that the proxy is up-to-date, more upgrade passes may be required.

Reply:
For `xoa-upgraded, updater-upgraded, installer-upgraded` i suggest `upgrading` message with a `spinner` as icon.

Reply:
Let's keep this simple for now, we'll fix it in the future if necessary.